subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/release/ReleasePlugin.java

```diff
@@ -55,49 +53,49 @@ public void apply(final Project project) {
-                t.setDescription("Tests the release procedure and cleans up. Safe to be invoked multiple times.");
-                //releaseCleanUp is already set up to run all his "subtasks" after performRelease is performed
-                //releaseNeeded is used here only to execute the code paths in the release needed task (extra testing)
-                t.getExecCommands().add(execCommand("Performing release in dry run, with cleanup",
-                    asList("./gradlew", RELEASE_NEEDED, PERFORM_RELEASE_TASK, RELEASE_CLEAN_UP_TASK, "-PdryRun")));
-                TaskSuccessfulMessage.logOnSuccess(t, "  The release test was successful. Ship it!");
-            }
+        TaskMaker.task(project, TEST_RELEASE_TASK, ShipkitExecTask.class, t -> {
+            t.setDescription("Tests the release procedure and cleans up. Safe to be invoked multiple times.");
+            //releaseCleanUp is already set up to run all his "subtasks" after performRelease is performed
+            //releaseNeeded is used here only to execute the code paths in the release needed task (extra testing)
+            t.getExecCommands().add(execCommand("Performing release in dry run, with cleanup",
+                asList("./gradlew", RELEASE_NEEDED, PERFORM_RELEASE_TASK, RELEASE_CLEAN_UP_TASK, "-PdryRun")));
+            TaskSuccessfulMessage.logOnSuccess(t, "  The release test was successful. Ship it!");
+        });
+
+        TaskMaker.task(project, RELEASE_CLEAN_UP_TASK, t -> {
+            t.setDescription("Cleans up the working copy, useful after dry running the release");
+
+            //using finalizedBy so that all clean up tasks run, even if one of them fails
+            t.finalizedBy(GitPlugin.PERFORM_GIT_COMMIT_CLEANUP_TASK);
+            t.finalizedBy(GitPlugin.TAG_CLEANUP_TASK);
         });
 
-        TaskMaker.task(project, RELEASE_CLEAN_UP_TASK, new Action<Task>() {
-            public void execute(final Task t) {
-                t.setDescription("Cleans up the working copy, useful after dry running the release");
+        TaskMaker.task(project, CONTRIBUTOR_TEST_RELEASE_TASK, ShipkitExecTask.class, t -> {
+            //modelled after testReleaseTask, see its code comments
+            t.setDescription("Similar to 'testRelease' but excludes tasks that require Git/Bintray permissions. " +
+                "Useful for contributors who don't have the permissions.");
 
-                //using finalizedBy so that all clean up tasks run, even if one of them fails
-                t.finalizedBy(GitPlugin.PERFORM_GIT_COMMIT_CLEANUP_TASK);
-                t.finalizedBy(GitPlugin.TAG_CLEANUP_TASK);
-            }
+            t.getExecCommands().add(execCommand("Performing release in dry run, with cleanup",
+                asList("./gradlew", RELEASE_NEEDED, PERFORM_RELEASE_TASK, RELEASE_CLEAN_UP_TASK, "-PdryRun",
+                    "-x", GitPlugin.GIT_PUSH_TASK, "-x", ShipkitBintrayPlugin.BINTRAY_UPLOAD_TASK)));
```

Comment:
Due to `-x bintrayUpload` this one will fail if we are running `contributorTestRelease` on a gradle plugin project (because we are using `publishPlugins` task in this case instead of `bintrayUpload`).
What about also covering this case?

Reply:
Great catch, thank you!!! How did you find out about this problem? I will try to figure out a solution to it.

Reply:
I had a similar problem recently. Therefore I was a little bit curious about it.